File: auction-app-backend/src/main/java/com/internship/auctionapp/requests/UpdateUserRequest.java
package com.internship.auctionapp.requests;

import com.internship.auctionapp.models.Address;
import com.internship.auctionapp.util.AuthenticationProvider;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import java.util.Date;

@Data
@NoArgsConstructor
public class UpdateUserRequest {
    @NotBlank
    private String firstName;

    @NotBlank
    private String lastName;

    @NotBlank
    private String email;

    private String phoneNumber;

    private String profileImageUrl;

    private Date dateOfBirth;

    private Address address;

    private AuthenticationProvider authenticationProvider;
}

File: auction-app-ui/src/models/request/update/updateUserRequest.ts
import { AuthenticationProvider } from 'models/enum/authenticationProvider';
import { CreateAddressRequest } from 'models/request/create/createAddressRequest';

export type UpdateUserRequest = {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  profileImageUrl: string | null;
  dateOfBirth: Date;
  address: CreateAddressRequest | null;
  authenticationProvider: AuthenticationProvider;
};
